e2e/terraform/provision-infra/userdata/windows-2022.ps1

```diff
@@ -0,0 +1,180 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+<powershell>
+
+Set-StrictMode -Version latest
+$ErrorActionPreference = "Stop"
+
+$RunningAsAdmin = ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator")
+if (!$RunningAsAdmin) {
+  Write-Error "Must be executed in Administrator level shell."
+  exit 1
+}
+
+# -------------------------------------------
+# Disks and Directories
+
+# Bring ebs volume online with read-write access
+Get-Disk | Where-Object IsOffline -Eq $True | Set-Disk -IsOffline $False
+Get-Disk | Where-Object isReadOnly -Eq $True | Set-Disk -IsReadOnly $False
+
+New-Item -ItemType Directory -Force -Path C:\opt -ErrorAction Stop
+New-Item -ItemType Directory -Force -Path C:\opt\nomad
+New-Item -ItemType Directory -Force -Path C:\etc\nomad.d
+New-Item -ItemType Directory -Force -Path C:\tmp
+New-Item -ItemType Directory -Force -Path C:\opt\consul
+New-Item -ItemType Directory -Force -Path C:\etc\consul.d
+
+# -------------------------------------------
+# Install Consul Agent
+
+Set-Location C:\opt
+
+Try {
+    $releases = "https://releases.hashicorp.com"
+    $version = "1.21.1+ent"
+    $url = "${releases}/consul/${version}/consul_${version}_windows_amd64.zip"
+
+    Write-Output "Downloading Consul from: $url"
+    Invoke-WebRequest -Uri $url -Outfile consul.zip -ErrorAction Stop
+    Expand-Archive .\consul.zip .\ -ErrorAction Stop
+    Move-Item consul.exe C:\opt\consul.exe -Force -ErrorAction Stop
+    C:\opt\consul.exe version
+    rm consul.zip
+
+    New-Service `
+      -Name "Consul" `
+      -BinaryPathName "C:\opt\consul.exe agent -config-dir C:\etc\consul.d" `
+      -StartupType "Automatic" `
+      -ErrorAction Ignore
+
+} Catch {
+    Write-Output "Failed to install Consul."
+    Write-Output $_
+    $host.SetShouldExit(-1)
+    throw
+}
+
+Write-Output "Installed Consul."
+
+# -------------------------------------------
+# Install service and firewall rules for Nomad
+# Note the service can't run until we upload Nomad too
+
+Try {
+    New-NetFirewallRule `
+      -DisplayName 'Nomad HTTP Inbound' `
+      -Profile @('Public', 'Domain', 'Private') `
+      -Direction Inbound `
+      -Action Allow `
+      -Protocol TCP `
+      -LocalPort @('4646')
```

Comment:
I suppose we don't need 4647 because the client makes that connection outbound to the server?